climada/hazard/base.py

```diff
@@ -1410,3 +1413,142 @@ def _read_att_excel(self, file_name, var_names):
+            Default is 100.
+
+        Returns
+        -------
+        haz_concat: climada.hazard.Hazard()
+            Concatenated hazard.
+
+        Raises
+        ------
+        ValueError
+        """
+
+        #Check units consistency among hazards
+        units = {haz.units for haz in haz_list if haz.units != ''}
+        if len(units) > 1:
+            raise ValueError("The haz_list contains hazards with different "
+                            "units %f. The hazards are incompatible and "
+                            "cannot be concatenated.", str(units))
+        elif len(units) == 0:
+            units = {''}
+
+        #Define comon centroids
+        if centroids is None:
+            centroids = Centroids().union(*[haz.centroids
+                                           for haz in haz_list])
+        #Define empty concatenate hazard
+        haz_concat = Hazard()
+        haz_concat.units = units.pop()
+        haz_concat.centroids = centroids
+
+        #Indices for mapping matrices onto common centroids
+        hazcent_in_cent_idx_list = [
+            u_coord.assign_coordinates(haz.centroids.coord, centroids.coord,
+                                       threshold=threshold)
+            for haz in haz_list
+            ]
+
+        if any([-1 in idx_list for idx_list in hazcent_in_cent_idx_list]):
+            raise ValueError("At least one hazard centroid is at a larger distance"
+                    "than the given threshold %f from the given centroids. "
+                    "To perform the concatenation please choose a larger "
+                    "threshold or enlarge the centroids", threshold)
+
+
+        #Concatenate attributes - hazards are assumed to have the same attributes
+        for attr_name in vars(haz_list[0]).keys():
+            attr_val_list = [getattr(haz, attr_name) for haz in haz_list]
+            if isinstance(attr_val_list[0], sparse.csr.csr_matrix):
+                #Map sparse matrix onto centroids.
+                matrix = (
+                    sparse.csr_matrix(
+                        (matrix.data, cent_idx[matrix.indices], matrix.indptr),
+                        shape=(matrix.shape[0], centroids.size)
+                        )
+                    for matrix, cent_idx in zip(attr_val_list, hazcent_in_cent_idx_list)
+                    )
+                setattr(haz_concat, attr_name, sparse.vstack(matrix, format='csr'))
+            elif (isinstance(attr_val_list[0], np.ndarray)
+                  and attr_val_list[0].ndim == 1):
+                setattr(haz_concat, attr_name, np.hstack(attr_val_list))
+            elif isinstance(attr_val_list[0], list):
+                setattr(haz_concat, attr_name, sum(attr_val_list, []))
+            elif isinstance(attr_val_list[0], TagHazard):
+                for tag in attr_val_list:
+                    if tag is not haz_concat.tag: haz_concat.tag.append(tag)
+
+        haz_concat.sanitize_event_ids()
```

Comment:
Is there a use case for providing a `sanitize_event_ids=True` parameter in this method, making this line optional (and instead throwing an error if the combined IDs aren't unique)? Would a user ever want event IDs to persist over the concatenation? (Or is the ID just an index variable, and they should use unique event names for this?)

Reply:
Hazard ids is just for internal indexing. The events are characterized by name and by date.